simulation/traffic_simulator/include/traffic_simulator/api/api.hpp

```diff
@@ -319,24 +324,10 @@ class API
 
 #undef FORWARD_TO_ENTITY_MANAGER
 
-  void requestSpeedChange(const std::string & name, double target_speed, bool continuous);
-
-  void requestSpeedChange(
-    const std::string & name, const double target_speed, const speed_change::Transition transition,
-    const speed_change::Constraint constraint, const bool continuous);
-
-  void requestSpeedChange(
-    const std::string & name, const speed_change::RelativeTargetSpeed & target_speed,
-    bool continuous);
-
-  void requestSpeedChange(
-    const std::string & name, const speed_change::RelativeTargetSpeed & target_speed,
-    const speed_change::Transition transition, const speed_change::Constraint constraint,
-    const bool continuous);
-
 private:
-  bool updateSensorFrame();
   bool updateEntityStatusInSim();
+  std::optional<traffic_simulator_msgs::msg::EntityStatus> updateEntityStatusInSim(
+    const std::string & entity_name, traffic_simulator_msgs::msg::EntityStatus status);
```

Comment:
```suggestion
  auto updateEntityStatusInSim(
    const std::string & entity_name, const traffic_simulator_msgs::msg::EntityStatus & status)
    -> std::optional<traffic_simulator_msgs::msg::EntityStatus>;
```

Reply:
I'm not able to apply this suggestion because in function there is assignment to `status` variable

Reply:
OK, I withdraw this change request.